Source/WebCore/style/MatchedDeclarationsCache.cpp

```diff
@@ -65,6 +65,9 @@ bool MatchedDeclarationsCache::isCacheable(const Element& element, const RenderS
     // Document::setWritingMode/DirectionSetOnDocumentElement. We can't skip the applying by caching.
     if (&element == element.document().documentElement())
         return false;
+    // FIXME: matchedPseudoElementIds should be PseudoElementIdentifier-aware instead.
+    if (!style.pseudoElementNameArgument().isNull())
+        return false;
```

Comment:
I think the FIXME is a very indirectly related to the actual problem.

As I understand the issue is that this hits `ASSERT(!parentStyle->getCachedPseudoStyle(pseudoElementIdentifier));` in `Element::resolvePseudoElementStyle` but the problem there is that getCachedPseudoStyle doesn't support pseudo elements with arguments so results are unexpected.

matchedPseudoElementIds may be part of solving that problem but currently getCachedPseudoStyle should probably be always returning nullptr for properties with argument.

Adding this test for now is ok for now too but not ideal. At least the FIXME should be improved.